### ISSUE TITLE
[Bug] Product variations not working

### ISSUE THREAD

**Issue by Crisis2010** [vuestorefront/shopware-pwa#1712]
### Describe the bug
Any product with variant causes 500 internal server error

### REPRODUCTION
1. Go to the following url of your demo instance.
https://shopware-pwa.storefrontcloud.io/93-Crop-Jeans/290980000.3

2. Start local prod/dev shopware instance with shopware-pwa-plugin 0.3.1 - dev-master

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

Latest stable (generated by CLI)

### Environment

```markdown
- OS: macOS
- Node: 14
- Shopware API: 6.4.5.1
- Shopware PWA version: 1.0.0-RC.3
```

### Relevant log output

```shell
{"errors":[{"code":"0","status":"500","title":"Internal Server Error","detail":"Notice: Undefined index: 43a23e0c03bf4ceabc6055a2185faa87","meta":{"trace":[{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php","line":388,"function":"loadOneToManyWithPagination","class":"Shopware\\Core\\Framework\\DataAbstractionLayer\\Dbal\\EntityReader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php","line":1067,"function":"loadOneToMany","class":"Shopware\\Core\\Framework\\DataAbstractionLayer\\Dbal\\EntityReader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php","line":126,"function":"fetchAssociations","class":"Shopware\\Core\\Framework\\DataAbstractionLayer\\Dbal\\EntityReader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php","line":93,"function":"_read","class":"Shopware\\Core\\Framework\\DataAbstractionLayer\\Dbal\\EntityReader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Profiling\/Entity\/EntityReaderProfiler.php","line":36,"function":"read","class":"Shopware\\Core\\Framework\\DataAbstractionLayer\\Dbal\\EntityReader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/System\/SalesChannel\/Entity\/SalesChannelRepository.php","line":177,"function":"read","class":"Shopware\\Core\\Profiling\\Entity\\EntityReaderProfiler","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/System\/SalesChannel\/Entity\/SalesChannelRepository.php","line":118,"function":"read","class":"Shopware\\Core\\System\\SalesChannel\\Entity\\SalesChannelRepository","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Content\/Product\/SalesChannel\/Detail\/ProductDetailRoute.php","line":120,"function":"search","class":"Shopware\\Core\\System\\SalesChannel\\Entity\\SalesChannelRepository","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Content\/Product\/SalesChannel\/Detail\/CachedProductDetailRoute.php","line":121,"function":"load","class":"Shopware\\Core\\Content\\Product\\SalesChannel\\Detail\\ProductDetailRoute","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/System\/SystemConfig\/SystemConfigService.php","line":349,"function":"Shopware\\Core\\Content\\Product\\SalesChannel\\Detail\\{closure}","class":"Shopware\\Core\\Content\\Product\\SalesChannel\\Detail\\CachedProductDetailRoute","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/Adapter\/Cache\/CacheTracer.php","line":39,"function":"trace","class":"Shopware\\Core\\System\\SystemConfig\\SystemConfigService","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/Adapter\/Translation\/Translator.php","line":100,"function":"Shopware\\Core\\Framework\\Adapter\\Cache\\{closure}","class":"Shopware\\Core\\Framework\\Adapter\\Cache\\CacheTracer","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/Adapter\/Cache\/CacheTracer.php","line":40,"function":"trace","class":"Shopware\\Core\\Framework\\Adapter\\Translation\\Translator","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/Adapter\/Cache\/CacheTagCollection.php","line":43,"function":"Shopware\\Core\\Framework\\Adapter\\Cache\\{closure}","class":"Shopware\\Core\\Framework\\Adapter\\Cache\\CacheTracer","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Framework\/Adapter\/Cache\/CacheTracer.php","line":41,"function":"trace","class":"Shopware\\Core\\Framework\\Adapter\\Cache\\CacheTagCollection","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/storefront\/Framework\/Cache\/CacheTracer.php","line":37,"function":"trace","class":"Shopware\\Core\\Framework\\Adapter\\Cache\\CacheTracer","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/storefront\/Theme\/ThemeConfigValueAccessor.php","line":53,"function":"Shopware\\Storefront\\Framework\\Cache\\{closure}","class":"Shopware\\Storefront\\Framework\\Cache\\CacheTracer","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/storefront\/Framework\/Cache\/CacheTracer.php","line":38,"function":"trace","class":"Shopware\\Storefront\\Theme\\ThemeConfigValueAccessor","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/Content\/Product\/SalesChannel\/Detail\/CachedProductDetailRoute.php","line":122,"function":"trace","class":"Shopware\\Storefront\\Framework\\Cache\\CacheTracer","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware-pwa\/shopware-pwa\/src\/Pwa\/PageLoader\/ProductPageLoader.php","line":80,"function":"load","class":"Shopware\\Core\\Content\\Product\\SalesChannel\\Detail\\CachedProductDetailRoute","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware-pwa\/shopware-pwa\/src\/Pwa\/Controller\/PageController.php","line":199,"function":"load","class":"SwagShopwarePwa\\Pwa\\PageLoader\\ProductPageLoader","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware-pwa\/shopware-pwa\/src\/Pwa\/Controller\/PageController.php","line":175,"function":"getPageResult","class":"SwagShopwarePwa\\Pwa\\Controller\\PageController","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":156,"function":"resolve","class":"SwagShopwarePwa\\Pwa\\Controller\\PageController","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":78,"function":"handleRaw","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/Kernel.php","line":199,"function":"handle","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpCache\/SubRequestHandler.php","line":85,"function":"handle","class":"Symfony\\Component\\HttpKernel\\Kernel","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpCache\/HttpCache.php","line":479,"function":"handle","class":"Symfony\\Component\\HttpKernel\\HttpCache\\SubRequestHandler","type":"::"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpCache\/HttpCache.php","line":269,"function":"forward","class":"Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpCache\/HttpCache.php","line":285,"function":"pass","class":"Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache","type":"-\u003E"},{"file":"\/sw6\/vendor\/symfony\/http-kernel\/HttpCache\/HttpCache.php","line":213,"function":"invalidate","class":"Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/HttpKernel.php","line":178,"function":"handle","class":"Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache","type":"-\u003E"},{"file":"\/sw6\/vendor\/shopware\/core\/HttpKernel.php","line":81,"function":"doHandle","class":"Shopware\\Core\\HttpKernel","type":"-\u003E"},{"file":"\/sw6\/public\/index.php","line":77,"function":"handle","class":"Shopware\\Core\\HttpKernel","type":"-\u003E"}],"file":"\/sw6\/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php","line":545}}]}
```

**Comment by patzick** [comment]
we updated the demo instance to the master branch so this problem doesn't occur anymore. It will be available in shopware API `6.4.6`. Thank you all for your involvement. Closing :)

**Comment by lbebnowicz** [comment]
Thank you for the info!
